python/postprocessing/framework/treeReaderArrayTools.py

```diff
@@ -21,31 +21,30 @@ def InputTree(tree,entrylist=None):
     tree.entries = tree._ttreereader.GetEntries(False)
     return tree
 
-def getArrayReader(tree, branchName, isClean=False):
-    """Make a reader for branch branchName containing a variable-length value array. 
-       If you are sure nobody has yet read from the tree, you can set isClean to True and save some overhead."""
+def getArrayReader(tree, branchName):
+    """Make a reader for branch branchName containing a variable-length value array."""
     if branchName not in tree._ttras:
        if not tree.GetBranch(branchName): raise RuntimeError, "Can't find branch '%s'" % branchName
        leaf = tree.GetBranch(branchName).GetLeaf(branchName)
        if not leaf.GetLen() == 0: raise RuntimeError, "Branch %s is not a variable-length value array" % branchName
        typ = leaf.GetTypeName()
-       tree._ttras[branchName] = _makeArrayReader(tree, typ, branchName, remakeAllFirst=not(isClean))
+       tree._ttras[branchName] = _makeArrayReader(tree, typ, branchName)
     return tree._ttras[branchName]
 
-def getValueReader(tree, branchName, isClean=False):
-    """Make a reader for branch branchName containing a single value. 
-       If you are sure nobody has yet read from the tree, you can set isClean to True and save some overhead."""
+def getValueReader(tree, branchName):
+    """Make a reader for branch branchName containing a single value."""
     if branchName not in tree._ttrvs:
        if not tree.GetBranch(branchName): raise RuntimeError, "Can't find branch '%s'" % branchName
        leaf = tree.GetBranch(branchName).GetLeaf(branchName)
        if not leaf.GetLen() == 1: raise RuntimeError, "Branch %s is not a value" % branchName
        typ = leaf.GetTypeName()
-       tree._ttrvs[branchName] = _makeValueReader(tree, typ, branchName, remakeAllFirst=not(isClean))
+       tree._ttrvs[branchName] = _makeValueReader(tree, typ, branchName)
     return tree._ttrvs[branchName]
 
 
 def readBranch(tree, branchName):
     """Return the branch value if the branch is a value, and a TreeReaderArray if the branch is an array"""
+    if tree._ttreereader._isClean: raise RuntimeError, "readBranch must not be called before setting the TTreeReader entry (or calling Next())"
```

Comment:
I think the error message here should tell the user to call `gotoEntry` (you don't want him to call `SetEntry` directly, behind your back; and, it even if he did, he'd still get this exception since you're checking `_isClean`)